snapcraft/internal/project_loader/grammar/_processor.py

```diff
@@ -73,6 +73,8 @@ def _parse_dict(section, statement, statements, project_options,
     from ._try import TryStatement
 
     for key, value in section.items():
+        value = value if isinstance(value, list) else {value}
```

Comment:
I was thinking we were going to need to rewrite some of the grammar to support strings, but treating them as iterables is clever. However, I'm not a huge fan of the inline if statements-- would you mind rewriting this to be a bit more verbose? Adding a comment about why this is required would also be useful in the future.

Reply:
Done.